kafka-producer-application/kafka/README.md

```diff
@@ -0,0 +1,166 @@
+      put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+      put(ProducerConfig.ACKS_CONFIG, "1");
+  }};
+  
+  Producer<String, String> producer = new KafkaProducer<>(properties);
+```
+
+For more about Kafka Producer Configurations, please [refer to the documentation](https://docs.confluent.io/platform/current/installation/configuration/producer-configs.html).
+
+To prepare an event to send, we create instances of `ProducerRecord` - which include the target topic, key and value, of the event
+to be sent to Kafka.
+
+```java annotate
+    final String key = "...";
+    final String value = "...";
+    final String topic = "...";
+    ProducerRecord<String, String> record = ProducerRecord<>(topic, key, value);
+```
+
+## Send, No Callback
+
+Given our `ProducerRecord`, let's send this record via our `Producer`. 
+
+```java annotate
+    producer.send(record);
+```
+
+## Send With Callback
+
+There are times when an application may want a deeper understanding of what occurred when an event was sent to Kafka. In this
+case we can utilize the implementation of the `send` method with a `Callback` function. 
+
+The `Callback` provides a way of handling any actions you want to take on request completion asynchronously. Note that the 
+`Callback` code executes on the producer’s I/O thread and any time consuming tasks could cause a delay in sending new 
+records, so any code here should be designed to execute quickly.
+
+Here is an example of a `Callback` - implemented as a lambda - that prints the offset and partition of the event sent to Kafka.
+
+```java annotate
+    Callback callback = (recordMetadata, e) -> 
+      System.out.println("topic: " + recordMetadata.topic() + 
+              "partition: , " + recordMetadata.partition() +
+              "offset: , " + recordMetadata.offset());
+
+    // use the callback with the `send` method...
+    producer.send(record, callback);
+```
+
+## Using Example
+
+<details>
+<summary>Execute the Unit Tests</summary>
+
+There are junit test cases in this repo, exercising examples of both implementations of the `send` method of `KafkaProducer`.
+
+To run the unit tests, use the provided Gradle Wrapper:
+
+```shell
+./gradlew clean :kafka-producer-application:kafka:test --info  
+```
+
+The results of the tests can be found in the `build/reports/index.html` report. Drill down using the links in the report,
+you should see the results of `KafkaProducerApplicationTest`:
+
+![Test Result](images/test-results.png)
+
+</details>
+
+<details>
+<summary>Run with Confluent Local</summary>
+
+You can run the example application in this tutorial using `confluent local`.
+
+### Prerequisites
+* [Confluent CLI](https://docs.confluent.io/confluent-cli/current/install.html) 
+* Docker running via [Docker Desktop](https://docs.docker.com/desktop/) or [Docker Engine](https://docs.docker.com/engine/install/)
+* [kcat](https://github.com/edenhill/kcat)
```

Comment:
We're moving towards having tutorials only introduce one thing, so maybe we can drop `kcat` and use the Confluent CLI to consume the records for verifications.  

Don't get me wrong, I think `kcat` is awesome, so we could add a separate tutorial for it.